runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1805,13 +1805,26 @@ Java_java_lang_Class_getNestHostImpl(JNIEnv *env, jobject recv)
 	J9Class *nestHost = clazz->nestHost;
 
 	if (NULL == nestHost) {
-		if (J9_VISIBILITY_ALLOWED == vmFuncs->loadAndVerifyNestHost(currentThread, clazz, J9_LOOK_NO_THROW)) {
-			nestHost = clazz->nestHost;
+		J9Class *clazzToUse = clazz;
+		J9JavaVM *vm = currentThread->javaVM;
+		if (J2SE_VERSION(vm) >= J2SE_V15) {
```

Comment:
Why not `#if JAVA_SPEC_VERSION >= 15` instead of runtime check?

Reply:
I've changed the code to use `#if JAVA_SPEC_VERSION >= 15`